stackslib/src/net/p2p.rs

```diff
@@ -500,6 +506,23 @@ impl PeerNetwork {
         network
     }
 
+    pub fn set_proposal_thread(&mut self, thread: JoinHandle<()>) {
```

Comment:
Should this check if `self.block_proposal_thread` is `None` first?

Reply:
I don't think so -- the caller needs to have already checked this, because it shouldn't even get a `JoinHandle` if the `block_proposal_thread` is still running.